src/messaging/ExchangeDelegate.h

```diff
@@ -100,5 +100,39 @@ class DLL_EXPORT ExchangeDelegate
     virtual ExchangeMessageDispatch & GetMessageDispatch() { return ApplicationExchangeDispatch::Instance(); }
 };
 
+/**
+ * @brief
+ *   This class handles unsolicited messages. The implementation can peek the message and creating a new exchange for it or not.
+ */
+class DLL_EXPORT UnsolicitedMessageHandler
+{
+public:
+    virtual ~UnsolicitedMessageHandler() {}
+
+    /**
+     * @brief
+     *   This function handles a unsolicited CHIP message.
+     *
+     *   If the implmentation returns CHIP_NO_ERROR, the message layer will create a new exchange with newDelegate, associated to
+     * the message, and using the new exchange to handle future messages.
+     *
+     *   The acceptor peek the message here, and the message payload will be sent to the new exchange regardless.
+     *
+     *  @param[in]  payloadHeader A reference to the PayloadHeader object.
+     *  @param[in]  payload       A handle to the PacketBuffer object holding the message payload.
+     *  @param[out] newDelegate   A new exchange delegate being used by the new exchange to handle future messages.
+     */
+    virtual CHIP_ERROR OnUnsolicitedMessageReceived(const PayloadHeader & payloadHeader, ExchangeDelegate *& newDelegate) = 0;
+
+    /**
+     * @brief
+     *   This function is called when OnUnsolicitedMessageReceived successfully returns a new delegate, but the session manager
+     *   fails to assign the delegate to a new exchange, then the delegate should be released to avoid resource leak.
+     *
+     *  @param[in] delegate   The exchange delegate to be released.
+     */
+    virtual void ReleaseDelegate(ExchangeDelegate * delegate) {}
```

Comment:
Perhaps:
```suggestion
    virtual void ExchangeCreationFailed(ExchangeDelegate * delegate) {}
```
?  Don't feel too strongly about this, but it highlights when exactly this is used.